### PR TITLE
 fix: implement configmap usage for centraldashboard links 

### PR DESCRIPTION
**Pull request by saffaalvi**
Resolves https://github.com/StatCan/kubeflow/issues/40

This fix also includes changing the configmap resource `centraldashboard-links-config` to our own links with translations, which was edited in the dev cluster.

### CODE REVIEW

File: components/centraldashboard/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "compilerOptions":{
         "module":"commonjs",
         "esModuleInterop":true,
-        "target":"es6",
+        "target":"es2019",
```
Comment (brendangadd) [inline components/centraldashboard/tsconfig.json:5]: Why is this necessary? Is it because of the `Object` methods being used? Could we implement in a way that doesn't require more recent JS support? It doesn't really matter for the server side, but it's a big deal for the browser.

File: components/centraldashboard/app/api.ts [hunk 2]
```diff
@@ -70,18 +69,42 @@ export class Api {
...
+          async (req: Request, res: Response) => {
+            const acceptlanguage = req.header('accept-language');
             const linksData = await this.k8sService.getDashboardLinks();
-            let links = {};
+            let langLinks = {};
             try {
-              links=JSON.parse(linksData.data["links"]);
+              const links = JSON.parse(linksData.data["links"]);
+              const lang = this.resolveLanguage
+              (
```
Comment (brendangadd) [inline components/centraldashboard/app/api.ts:79]: Keep parenthesis on same line as method ref so it doesn't look like we're just referencing a property of `this`.

File: components/centraldashboard/app/api.ts [hunk 3]
```diff
@@ -70,18 +69,42 @@ export class Api {
...
+
+  getBrowserLanguages(lang: string) {
+    const languages = lang.split(',');
+    if (!languages) {
+      return [];
+    }
+    // Append fallbacks not explicit in browser languages.
+    // Non-destructive: string keys will be reported back in order of insertion.
+    let languagelist: {[code: string]: boolean};
+    languagelist = Object.fromEntries([].concat(languages).map(lang => [lang.split(/-|_|;/)[0], true]));
```
Comment (brendangadd) [inline components/centraldashboard/app/api.ts:107]: Good call on adding `;`. MDN doesn't show underscore as being a valid character for this header. Do some browsers use it in place of hyphen?

**Review by brendangadd — CHANGES_REQUESTED** [review]

File: components/centraldashboard/tsconfig.json (hunk 1 above)
Comment (saffaalvi) [reply components/centraldashboard/tsconfig.json:5]: `Object.fromEntries()` and `includes()` aren't in es6 but are both in es2019

File: components/centraldashboard/app/api.ts (hunk 3 above)
Comment (saffaalvi) [reply components/centraldashboard/app/api.ts:107]: No, accidentally left that in there. Thanks!

File: components/centraldashboard/app/api.ts [hunk 4]
```diff
@@ -93,18 +92,17 @@ export class Api {
...
+    if (!acceptlanguage) {
       return [];
     }
+    const languages = acceptlanguage.split(',');
     // Append fallbacks not explicit in browser languages.
     // Non-destructive: string keys will be reported back in order of insertion.
-    let languagelist: {[code: string]: boolean};
-    languagelist = Object.fromEntries([].concat(languages).map(lang => [lang.split(/-|_|;/)[0], true]));
-    return (Object.keys(languagelist));
+    const languagelist = Array.from(new Set(languages.map(lang => lang.split(/-|;/)[0])));
```
Comment (brendangadd) [inline components/centraldashboard/app/api.ts:105]: :star_struck:

**Review by brendangadd — APPROVED** [review]
